app/ante/bypass_ibc_fee_decorator.go

```diff
@@ -13,19 +17,29 @@ type anteHandler interface {
 
 type BypassIBCFeeDecorator struct {
 	nextAnte anteHandler
+	dk       distrkeeper.Keeper
+	sk       seqkeeper.Keeper
 }
 
-func NewBypassIBCFeeDecorator(nextAnte anteHandler) BypassIBCFeeDecorator {
-	return BypassIBCFeeDecorator{nextAnte: nextAnte}
+func NewBypassIBCFeeDecorator(nextAnte anteHandler, dk distrkeeper.Keeper, sk seqkeeper.Keeper) BypassIBCFeeDecorator {
+	return BypassIBCFeeDecorator{
+		nextAnte: nextAnte,
+		dk:       dk,
+		sk:       sk,
+	}
 }
 
 // SKIP FEE DEDUCT and MIN GAS PRICE Ante handlers for IBC relayer messages
 func (n BypassIBCFeeDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate bool, next sdk.AnteHandler) (sdk.Context, error) {
-	// ======== HACK ================
-	if isIBCRelayerMsg(tx.GetMsgs()) {
+	// ================ HACK ================
+	whitelisted, err := n.isWhitelistedRelayer(ctx, tx.GetMsgs())
+	if err != nil {
+		return ctx, fmt.Errorf("is whitelisted relayer: %w", err)
```

Comment:
I don't think we should fail the flow in any case as whitelist relayer is an optimization. worst case we fallback to paid flow.